open-apparel-registry/open-apparel-registry
# Add facilities vector tile endpoint along with vector tile map & vector tile layer React components

## Overview

This PR adds a vector tie endpoint to the Django app along with VectorTileFacilitiesMap and VectorTileFacilitiesLayer components to the React app. It also adds migrations to:

- create a `vector_tile` feature flag for Django-Waffle
- add an index on the `updated_at` column in the facilities table, since we're going to use the maximum value of that column as a tile cache key

Connects #722 

## Demo

![Screen Shot 2019-08-16 at 6 07 29 PM](https://user-images.githubusercontent.com/4165523/63200936-2197ef00-c051-11e9-9fe8-d2131c87c7e4.png)
![Screen Shot 2019-08-16 at 6 07 48 PM](https://user-images.githubusercontent.com/4165523/63200937-2197ef00-c051-11e9-99ed-496014177030.png)

## Notes

The `/tile` endpoint currently has `AllowAny` permissions set, but we want to adjust that to restrict things so that only the React app can use the tile endpoint. I'll create a new card for this.

## Testing Instructions

- serve this branch, `resetdb`, then `./scripts/manage waffle_switch vector_tile on`
- `./scripts/server` and visit the app on 6543 when it loads
- verify that you see the vector tile layer as depicted above and that the sets of facilities returned from the endpoint change on changing the filters and completing a search

## Checklist
